train/train.py

```diff
@@ -284,13 +308,50 @@ def train():
 
     random.shuffle(raw_data)
 
-    dataset = CustomDataset(raw_data, tokenizer)
+    if training_args.do_eval:
+        # Take 90:10 train-validation split
+        raw_train_data, raw_eval_data = split_data(raw_data, data_args.data_path, 0.9)
+        train_dataset = CustomDataset(raw_train_data, tokenizer)
+        eval_dataset = CustomDataset(raw_eval_data, tokenizer)
+    else:
+        train_dataset = CustomDataset(raw_data, tokenizer)
+        
+    def preprocess_logits_for_metrics(logits, labels):
+        pred_ids = torch.argmax(logits, dim=-1)
+        shift_logits = logits[..., :-1, :].contiguous()
+        shift_labels = labels[..., 1:].contiguous()
+        loss_fct = CrossEntropyLoss(reduction="none")
+        shift_logits = shift_logits.view(-1, tokenizer.vocab_size)
+        shift_labels = shift_labels.view(-1)
+        loss = loss_fct(shift_logits, shift_labels)
+        loss = torch.mean(loss.view(logits.shape[0], -1), dim=-1)
+        return pred_ids, loss
+    
+    def compute_metrics(eval_preds):
```

Comment:
Let's use appropriate blank spaces here too please. Also What is B x L ? Let's clarify them too.